### ISSUE TITLE
GPU build on ARM-64 failing

### ISSUE THREAD

**Issue by sdeoras** [tensorflow/tensorflow#27931]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Ubuntu 18.04 running on Nvidia Jetson Nano board
Linux jetson-01 4.9.140-tegra #1 SMP PREEMPT Wed Mar 13 00:32:22 PDT 2019 aarch64 aarch64 aarch64 GNU/Linux

- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
N/A

- TensorFlow installed from (source or binary):
source

- TensorFlow version:
v1.13.1

- Python version:
Python 2.7.15rc1

- Installed using virtualenv? pip? conda?:
N/A

- Bazel version (if compiling from source):
```
INFO: Invocation ID: e7fbbd06-6d2d-4fee-9260-a96c2efc12e9
Build label: 0.21.0- (@non-git)
Build target: bazel-out/aarch64-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Apr 17 11:17:18 2019 (1555499838)
Build timestamp: 1555499838
Build timestamp as int: 1555499838

```
- GCC/Compiler version (if compiling from source):
gcc (Ubuntu/Linaro 7.3.0-27ubuntu1~18.04) 7.3.0

- CUDA/cuDNN version:
cuda 10, cuDNN 7

- GPU model and memory:
Tegra based GPU on Jetson Nano bord

**Describe the problem**
I am trying to build TensorFlow on ARM64 (Jetson Nano board) for use with GPU. Build is failing.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

- Enabled 6GB swap space per: https://www.jetsonhacks.com/2019/04/14/jetson-nano-use-more-memory/
- Built bazel 0.21 from source.
- Checked out v1.13.1 from TF repo
- Applied these changes to build files (diff attached below): https://github.com/tensorflow/tensorflow/issues/21852#issuecomment-477885516
- ran `./configure` saying no to most items except yes for CUDA support

```
Ran following command to start the build
nohup bazel build -c opt --jobs 1 --local_resources 5000,1.0,1.0 --verbose_failures --config=opt --config=cuda --config=noaws --config=nogcp --config=nohdfs --config=noignite --config=nokafka //tensorflow:libtensorflow.so &

```
**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
[nohup.txt](https://github.com/tensorflow/tensorflow/files/3091084/nohup.txt)

[tf_jetson_nano_build.diff.txt](https://github.com/tensorflow/tensorflow/files/3092069/tf_jetson_nano_build.diff.txt)

**Comment by wdirons** [comment]
I believe this is the same problem reported in issue https://github.com/tensorflow/tensorflow/issues/25323

**Comment by sdeoras** [comment]
As discussed in #25323 I tried building with gcc-8 instead of gcc-7, however, that produced another error:
$ gcc --version
`gcc (Ubuntu 8.2.0-1ubuntu2~18.04) 8.2.0`
$ bazel build -c opt --verbose_failures --config=opt --config=noaws --config=nogcp --config=nohdfs --config=noignite --config=nokafka //tensorflow:libtensorflow.so
```
INFO: Invocation ID: b9c93771-fc51-4edb-b485-09a27f7878fb
INFO: Build option --define has changed, discarding analysis cache.
INFO: Analysed target //tensorflow:libtensorflow.so (0 packages loaded, 8857 targets configured).
INFO: Found 1 target...
ERROR: /home/sdeoras/.cache/bazel/_bazel_sdeoras/6333fd39180a43710cc9e4529b1411f9/external/double_conversion/BUILD.bazel:12:1: undeclared inclusion(s) in rule '@double_conversion//:double-conversion':
this rule is missing dependency declarations for the following files included by 'external/double_conversion/double-conversion/cached-powers.cc':
  '/usr/lib/gcc/aarch64-linux-gnu/8/include-fixed/limits.h'
  '/usr/lib/gcc/aarch64-linux-gnu/8/include-fixed/syslimits.h'
  '/usr/lib/gcc/aarch64-linux-gnu/8/include/stddef.h'
  '/usr/lib/gcc/aarch64-linux-gnu/8/include/stdarg.h'
  '/usr/lib/gcc/aarch64-linux-gnu/8/include/stdint.h'
Target //tensorflow:libtensorflow.so failed to build
INFO: Elapsed time: 26.678s, Critical Path: 11.99s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

**Comment by wdirons** [comment]
In between builds did you clear the bazel cache?

if not I would try:
```
bazel clean --expunge_async
rm -rf /home/sdeoras/.cache/
```

**Comment by sdeoras** [comment]
I think gcc-8 or higher is not going to work when building on Jetson Nano per this error:

ERROR: /home/sdeoras/.cache/bazel/_bazel_sdeoras/6333fd39180a43710cc9e4529b1411f9/external/nccl_archive/BUILD.bazel:151:1: error while parsing .d file: /home/sdeoras/.cache/bazel/_bazel_sdeoras/6333fd39180a43710cc9e4529b1411f9/execroot/org_tensorflow/bazel-out/aarch64-opt/bin/external/nccl_archive/_objs/nccl/rings.cu.pic.d (No such file or directory)
In file included from /usr/local/cuda-10.0/bin/../targets/aarch64-linux/include/cuda_runtime.h:83,
                 from <command-line>:
/usr/local/cuda-10.0/bin/../targets/aarch64-linux/include/crt/host_config.h:129:2: error: #error -- unsupported GNU version! gcc versions later than 7 are not supported!
 #error -- unsupported GNU version! gcc versions later than 7 are not supported!

**Comment by sdeoras** [comment]
I am wondering if there is a recommendation on a branch or commit I can try building using gcc-7.

**Comment by wdirons** [comment]
Reading the other defect you could try any of these options:

1) Removing `-c opt` and adding `--config=nativeopt` from the bazel command line
2) GCC 7.4
3) Building the v1.12.0 branch

**Comment by sdeoras** [comment]
@wdirons thanks so much for your support. I would love to try it out, however wanted to point out that v1.12.0 most likely wont work with cuda 10 and I don't have option to downgrade to cuda 9 on Jetson Nano.

**Comment by sdeoras** [comment]
@wdirons: Thanks for your inputs. v1.12.0 built just fine with cuda 10 on Jetson Nano using gcc 7.3. Please feel free to close this case since the gcc 7.3 related issue with v1.13.1 is being addressed at #25323

**Comment by muddham** [comment]
@sdeoras Closing this issue since its resolved. Thanks!
